pkg/inventory/account.go

```diff
@@ -54,7 +56,7 @@ func ForAccounts(existing []v1.ServiceAccount, desired []v1.ServiceAccount) Acco
 func accountMap(deps []v1.ServiceAccount) map[string]v1.ServiceAccount {
 	m := map[string]v1.ServiceAccount{}
 	for _, d := range deps {
-		m[d.Name] = d
+		m[fmt.Sprintf("%s.%s", d.Namespace, d.Name)] = d
```

Comment:
This kind of change isn't strictly required, as the data it receives *should* be part of the same namespace. However, it's more accurate to use the namespace in the key, as "tenant1.simplest" shouldn't be seen as the same as "tenant2.simplest". This fix helped me spot the real cause for the bug.